Example/Example/AnimationPreviewView.swift

```diff
@@ -54,12 +70,29 @@ struct AnimationPreviewView: View {
     .navigationTitle(animationSource.name.components(separatedBy: "/").last!)
     .frame(maxWidth: .infinity, maxHeight: .infinity)
     .background(Color.secondaryBackground)
+    .onReceive(timer) { _ in
+      updateIndex()
+    }
   }
 
   // MARK: Private
 
+  /// Used for remote animations only, when more than one URL is provided we loop over the urls every 2 seconds.
+  private let timer = Timer.publish(every: 2, on: .main, in: .common).autoconnect()
+  private let urls: [URL]
+
   @State private var animationPlaying = true
   @State private var sliderValue: AnimationProgressTime = 0
+  @State private var currentURLIndex: Int
+
+  private var fetchTrigger: Binding<AnyHashable> {
+    switch animationSource {
+    case .local:
+      return .constant(currentURLIndex)
+    case .remote:
+      return $currentURLIndex.mapGetter(transform: AnyHashable.init)
```

Comment:
I wonder if using `func loadAnimationTrigger<T: Hashable>(_ binding: Binding<Hashable>) -> Self)` instead of `AnyHashable` would let us avoid having to map this binding to an `AnyHashable` in the consumer code and instead let us pass in a `Binding<Int>`. That would be a nice callsite simplification.